### PR TITLE
EMF model for abapGit repositories view

### PR DESCRIPTION
**Pull request by shubhamWaghmare-sap**

### CODE REVIEW

File: org.abapgit.adt.backend/src/org/abapgit/adt/backend/model/abapgitrepositories/impl/RepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,1066 @@
...
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getStatusText <em>Status Text</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getRemoteUser <em>Remote User</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getRemotePassword <em>Remote Password</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getFirstCommit <em>First Commit</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getTransportRequest <em>Transport Request</em>}</li>
+ * </ul>
+ *
+ * @generated
+ */
+public class RepositoryImpl extends MinimalEObjectImpl.Container implements IRepository {
```
Comment (abinbaby-sap) [inline org.abapgit.adt.backend/src/org/abapgit/adt/backend/model/abapgitrepositories/impl/RepositoryImpl.java:55]: Provide an implementation for the **equals and hashcode** method for the Repository object.
Also, do the same for other objects like AbapObjectImpl...

File: test/org.abapgit.adt.backend.test/src/org/abapgit/adt/backend/internal/TestsUnitRepositories.java [hunk 2]
```diff
@@ -4,23 +4,28 @@
 
 import java.util.List;
 
-import org.abapgit.adt.backend.IRepository;
+import org.abapgit.adt.backend.model.abapgitrepositories.IRepositories;
+import org.abapgit.adt.backend.model.abapgitrepositories.IRepository;
+import org.abapgit.adt.backend.model.abapgitrepositories.impl.AbapgitrepositoriesFactoryImpl;
 import org.junit.Test;
 
 public class TestsUnitRepositories {
```
[the remark this answers is not in the document]
Comment (shubhamWaghmare-sap) [reply test/org.abapgit.adt.backend.test/src/org/abapgit/adt/backend/internal/TestsUnitRepositories.java]: Yes this test is irrelevant. Removed the test.
Also, APACK handling has not been changed to EMF-based handling.

File: org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/AbapGitObjLogService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	public List<IAbapObject> listChildObjects(IAbapObject node) {
+		return node.getAbapLogObjectChildren();
+	}
+
+	public IAbapObjects renderObjLogInput(IAbapObjects deserializedAbapObjects) {
+
+		IAbapObjects ObjLogInput = IAbapObjectsFactory.eINSTANCE.createAbapObjects();
+		IAbapObject newObj_type = IAbapObjectsFactory.eINSTANCE.createAbapObject();
```
[the remark this answers is not in the document]
Comment (shubhamWaghmare-sap) [reply org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/AbapGitObjLogService.java]: Made the requested change here.

File: org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/AbapGitObjLogService.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+		int m = deserializedAbapObjects.getAbapObjects().size();
+
+		for (int j = 0; j < m; j++) {
+
+			IAbapObject abapobject = deserializedAbapObjects.getAbapObjects().get(j);
```
[the remark this answers is not in the document]
Comment (shubhamWaghmare-sap) [reply org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/AbapGitObjLogService.java]: Incorporated the suggestion

File: org.abapgit.adt.backend/model/abapgitrepositories.genmodel [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/deserializedAt"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/deserializedEmail"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/deserializedBy"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/status"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/statusText"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/remoteUser"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/remotePassword"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/firstCommit"/>
+      <genFeatures createChild="false" ecoreFeature="ecore:EAttribute abapgitrepositories.ecore#//Repository/transportRequest"/>
```
[the remark this answers is not in the document]
Comment (shubhamWaghmare-sap) [reply org.abapgit.adt.backend/model/abapgitrepositories.genmodel:27]: Not required. Deleted it from the EMF model.

File: org.abapgit.adt.backend/src/org/abapgit/adt/backend/model/abapgitrepositories/impl/RepositoryImpl.java (hunk 1 above)
Comment (shubhamWaghmare-sap) [reply org.abapgit.adt.backend/src/org/abapgit/adt/backend/model/abapgitrepositories/impl/RepositoryImpl.java:49]: Provided the hashcode and equals method.

**Pull request merged**